src/visual.ts

```diff
@@ -733,13 +760,19 @@ module powerbi.extensibility.visual {
 
             this.columsAndAxesTransform(maxWidthOfVerticalAxisLabel);
 
-            this.updateWidthOfColumn();
+            this.updateWidthOfColumn(isXaxisBordersOverriden ? UpdateColumnsWidthMode.markOnly : UpdateColumnsWidthMode.standardCalculation);
 
             this.createScales();
 
             this.applySelectionStateToData();
 
             this.render();
+
+            // The second rendering is necessary to calculate the width of rendered columns with counting of offset (when start and end x axis is set)
+            if (isXaxisBordersOverriden) {
+                this.updateWidthOfColumn(UpdateColumnsWidthMode.calculatePointsDiff);
+                this.render();
```

Comment:
Do we need to call render method twice?

Reply:
I haven't found another approach for a while. However, I can spend much time to look for another one.

Reply:
I found a new approach an will send it soon.